types/router.d.ts

```diff
@@ -28,7 +28,7 @@ export declare class VueRouter {
   go (n: number): void;
   back (): void;
   forward (): void;
-  getMatchedComponents (to?: RawLocation | Route): Component[];
+  getMatchedComponents (to?: RawLocation | Route): (ComponentOptions<Vue> | typeof Vue)[];
```

Comment:
```suggestion
  getMatchedComponents (to?: RawLocation | Route): ComponentOptions<Vue>[];
```

it will never be Vue, so we can remove that as well

Reply:
is `Vue.extend({})` never come here?
I thought ExtentedVue will come here.
like this.

```vue:component
<template>
  <div></div>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({});
</script>
```

```js:router
import Vue from 'vue'
import Router from 'vue-router'

Vue.use(Router)

export const createRouter = () => {
  return new Router({
    mode: 'history',
    routes: [
      { path: '/', component: () => import('@/pages/Index.vue') },
    ]
  })
}
```

Reply:
I just try pushed code as said.

Reply:
I totally overlooked the `Vue.extend` version of pages. The current implementation is correct, as we cannot transform any component option, so we leave them as they are.
Sorry about that